includes/Cmm.h:326:10: fatal error: 'DerivedConstants.h' file not found

Using the latest `shake-build`, `ba41ec6`, it ends with
```
Reading package info from "rts/stage1/package.conf.inplace" ... done.
rts-1.0: cannot find any of ["libHSrts.a","libHSrts.p_a","libHSrts-ghc8.1.20160104.so","libHSrts-ghc8.1.20160104.dylib","HSrts-ghc8.1.20160104.dll"] on library path (ignoring)
Reading rts/stage1/package-data.mk...
Reading dependencies from rts/stage1/build/.dependencies...

/----------------------------------------------\
| Run GccM Stage1 (package = rts)              |
|     input: rts/Updates.cmm                   |
| => output: rts/stage1/build/Updates.cmm.deps |
\----------------------------------------------/
In file included from rts/Updates.cmm:14:
includes/Cmm.h:326:10: fatal error: 'DerivedConstants.h' file not found
#include "DerivedConstants.h"
         ^
1 error generated.
shakeArgsWith    0.001s    0%                           
Function shake   0.027s    0%                           
Database read    0.005s    0%                           
With database    0.000s    0%                           
Running rules  150.147s   99%  =========================
Total          150.181s   99%                           
Error when running Shake build system:
* rts/stage1/build/libHSrts.a
* rts/stage1/build/stage1/build/AutoApply.o
* OracleQ (DependenciesKey ("rts/stage1/build/.dependencies","rts/stage1/build/stage1/build/AutoApply.o"))
* rts/stage1/build/.dependencies
* rts/stage1/build/Updates.cmm.deps
user error (Development.Shake.cmd, system command failed
Command: /usr/bin/clang -E -MM -Iincludes -Iincludes/dist-derivedconstants/header -Irts/stage1/build -Irts/stage1/build/autogen -MF rts/stage1/build/Updates.cmm.deps -MT rts/stage1/build/Updates.o -x c rts/Updates.cmm -Irts -Irts/stage1/build -DRtsWay="rts_v"
Exit code: 1
Stderr:
In file included from rts/Updates.cmm:14:
includes/Cmm.h:326:10: fatal error: 'DerivedConstants.h' file not found
#include "DerivedConstants.h"
         ^
1 error generated.
)
```

--- a/src/Rules/Generate.hs
+++ b/src/Rules/Generate.hs
@@ -69,10 +69,15 @@ compilerDependencies stage =
        , "primop-vector-tycons.hs-incl"
        , "primop-vector-tys.hs-incl" ]
 
+integerGmpDependencies :: [FilePath]
+integerGmpDependencies = ((pkgPath integerGmp -/- "gmp") -/-) <$>
+    [ "gmp.h" ] -- identical to integerGmpLibraryH, but doesn't require the import.
+
 generatedDependencies :: Stage -> Package -> [FilePath]
 generatedDependencies stage pkg
     | pkg == compiler = compilerDependencies stage
     | pkg == rts = derivedConstantsDependencies
+    | pkg == integerGmp = integerGmpDependencies
     | stage == Stage0 = defaultDependencies
     | otherwise = []
 

--- a/src/Rules/Generate.hs
+++ b/src/Rules/Generate.hs
@@ -72,6 +72,7 @@ compilerDependencies stage =
 generatedDependencies :: Stage -> Package -> [FilePath]
 generatedDependencies stage pkg
     | pkg == compiler = compilerDependencies stage
+    | pkg == rts = derivedConstantsDependencies
     | stage == Stage0 = defaultDependencies
     | otherwise = []
 

--- a/src/Settings/Packages/Rts.hs
+++ b/src/Settings/Packages/Rts.hs
@@ -49,6 +49,13 @@ rtsPackageArgs = package rts ? do
           [ arg "-Irts"
           , arg $ "-I" ++ path -/- "build"
           , arg $ "-DRtsWay=\"rts_" ++ show way ++ "\""
+          -- rts **must** be compiled with optimizations. The INLINE_HEADER macro,
+          -- requires that functions are inlined to work as expected.  Inlining
+          -- only happens for optimized builds. Otherwise we can assume that
+          -- there is a non-inlined variant to use instead. But rts does not
+          -- provide non-inlined alternatives and hence needs the function to
+          -- be inlined. See also Issue #90
+          , arg $ "-O2"
 
           , (file "//RtsMessages.*" ||^ file "//Trace.*") ?
             arg ("-DProjectVersion=" ++ quote projectVersion)

